x/ccv/provider/keeper/grpc_query.go

```diff
@@ -182,3 +182,30 @@ func (k Keeper) QueryProposedConsumerChainIDs(goCtx context.Context, req *types.
 		ProposedChains: chains,
 	}, nil
 }
+
+func (k Keeper) QueryAllPairsValConAddrByConsumerChainID(goCtx context.Context, req *types.QueryAllPairsValConAddrByConsumerChainIDRequest) (*types.QueryAllPairsValConAddrByConsumerChainIDResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if req.ChainId == "" {
+		return nil, status.Error(codes.InvalidArgument, "empty chainId")
+	}
+
+	// list of pairs valconsensus addr <providerValConAddrs : consumerValConAddrs>
+	pairValConAddrs := []*types.PairValConAddrProviderAndConsumer{}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	validatorConsumerAddrs := k.GetAllValidatorsByConsumerAddr(ctx, &req.ChainId)
+	for _, data := range validatorConsumerAddrs {
+		pubKey, _, _ := k.GetKeyAssignmentReplacement(ctx, req.ChainId, types.ProviderConsAddress{Address: data.ProviderAddr})
```

Comment:
This is not the way to get the consumer key. Note that 
```
// GetKeyAssignmentReplacement returns the previous assigned consumer key and the current power
// for a provider validator for which a key assignment was received in this block. Both are
// needed to update the validator's power on the consumer chain at the end of the current block.
```
This means that most of the time there will be no key. 

Use `GetValidatorConsumerPubKey` instead. See `QueryValidatorConsumerAddr()` for an example of how to get the consumer key from the provider address, i.e.,
```golang
	providerAddrTmp, err := sdk.ConsAddressFromBech32(req.ProviderAddress)
	if err != nil {
		return nil, err
	}
	providerAddr := types.NewProviderConsAddress(providerAddrTmp)

	consumerKey, found := k.GetValidatorConsumerPubKey(ctx, req.ChainId, providerAddr)
	if !found {
		return &types.QueryValidatorConsumerAddrResponse{}, nil
	}
```

Reply:
Updated it, plz check again @mpoke